pandas/tseries/tests/test_daterange.py

```diff
@@ -477,13 +477,27 @@ def test_range_closed(self):
             closed = date_range(begin, end, closed=None, freq=freq)
             left = date_range(begin, end, closed="left", freq=freq)
             right = date_range(begin, end, closed="right", freq=freq)
+            expected_left = left
+            expected_right = right
 
-            expected_left = closed[:-1]
-            expected_right = closed[1:]
+            if end == closed[-1]:
+                expected_left = closed[:-1]
+            if begin == closed[0]:
+                expected_right = closed[1:]
 
             self.assertTrue(expected_left.equals(left))
             self.assertTrue(expected_right.equals(right))
 
+    def test_range_closed_boundary(self):
+        # GH 11804
+        closed_right = date_range('2015-09-12', '2015-12-01', freq='QS-MAR', closed='right')
+        closed_left = date_range('2015-09-01', '2015-10-30', freq='QS-MAR', closed='left')
+        expected_left = DatetimeIndex(['2015-09-01'])
+        expected_right = DatetimeIndex(['2015-12-01'])
```

Comment:
also add in ``closed=None``.

Further do this test for start/end on the boundaries & not on the boundaries.

I think you should have 9 possibilities here (3 from closed parm, 3 from if start/end on boundary)